### DISCUSSION TITLE
[A/L] (2021/22) Logica - Esercizio 1

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#327] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152324150-9fd51db1-907c-4b7f-945a-6effc09490a5.png)

**Comment by CiottoloMaggico** [comment]
![immagine](https://user-images.githubusercontent.com/79638284/217773603-d6ccfd8b-ee31-40c5-b981-43ec5ce48bd2.png)

**Comment by Elia-Belli** [comment]
Concordo con @CiottoloMaggico tranne che sulla $5$.

- $5$ **Vera**: riscriviamola con soli connettivi logici $\lnot ((\lnot A \lor B) \land (\lnot A \lor C)) \lor (\lnot A \lor (B \lor C))$ e semplifichiamo
$(A \land \lnot B) \lor (A \land \lnot C) \lor \lnot A \lor B \lor C$, questa proposizione è una tautologia (vedere la tabella) allora la $5$ è vera:
A B C Y
0 0 0 1
0 0 1 1
0 1 0 1
0 1 1 1
1 0 0 1
1 0 1 1
1 1 0 1 
1 1 1 1

**Comment by LuginiAndrea** [comment]
La 12 è vera. Sappiamo infatti che $A_1,...,A_n \models B$ equivale a chiedersi $A_1 \cdot ... \cdot A_n \cdot \neg B \in$ Unsat
Riscriviamo l'espressione che abbiamo come espressione booleana (sostituiamo le implicazioni $A \implies B$ con $\neg A + B$)
$$A \times \left(\overline{A} + B\right) \times \overline{\left(\overline{B} + C\right)} \times \overline{D}$$
Si nota subito come possiamo ignorare la moltiplicazione per $A$ negato in quanto UnSat per definizione
Ne segue che rimane
$$A \times B \times B \times \overline{C} \times \overline{D} = A \times B \times \overline{C} \times \overline{D}$$
Come detto sopra, per vedere se l'espressione iniziale modella $D$ equivale a chiederci se l'espressione ottenuta da questi calcoli è $\in$ Unsat
